### PR TITLE
feat(typography): Support custom properties in mdc-typography mixin

### PR DESCRIPTION
**Pull request by pndewit**
Apply all styles from `$mdc-typography-styles` instead of just a select set of properties.

This allows users to override `$mdc-typography-styles` and add custom properties which will then be used everywhere. For example, adding `color` or a different `font-family`.

### CODE REVIEW

File: packages/mdc-typography/_mixins.scss [hunk 1]
```diff
@@ -31,12 +31,13 @@
...
-  font-weight: #{map-get($style-props, font-weight)};
-  letter-spacing: map-get($style-props, letter-spacing);
-  line-height: map-get($style-props, line-height);
-  text-decoration: map-get($style-props, text-decoration);
-  text-transform: map-get($style-props, text-transform);
+  @if not $adjust-margin {
+    $style-props: map_remove($style-props, margin);
+  }
+
+  @each $style, $value in $style-props {
```
Comment (acdvorak) [inline packages/mdc-typography/_mixins.scss]: `$style` is already being used by a param; maybe rename this to `$prop`?

File: packages/mdc-typography/_mixins.scss [hunk 2]
```diff
@@ -31,12 +31,13 @@
...
   @include mdc-typography-base;
 
-  font-size: map-get($style-props, font-size);
-  font-weight: #{map-get($style-props, font-weight)};
-  letter-spacing: map-get($style-props, letter-spacing);
-  line-height: map-get($style-props, line-height);
-  text-decoration: map-get($style-props, text-decoration);
-  text-transform: map-get($style-props, text-transform);
+  @if not $adjust-margin {
+    $style-props: map_remove($style-props, margin);
```
Comment (acdvorak) [inline packages/mdc-typography/_mixins.scss]: Our convention is to use underscores for Sass function names: `map-remove`

File: packages/mdc-typography/_mixins.scss [hunk 3]
```diff
@@ -22,7 +22,7 @@
   -webkit-font-smoothing: antialiased;
 }
 
-@mixin mdc-typography($style) {
+@mixin mdc-typography($style, $adjust-margin: false) {
```
Comment (acdvorak) [inline packages/mdc-typography/_mixins.scss]: Please update the signature of `mdc-typography` in README.md (under ["Sass Variables and Mixins"](https://github.com/pndewit/material-components-web/blob/86ac580c9eb8280c1622b5ea0efc0f933aaca3f4/packages/mdc-typography/README.md#sass-variables-and-mixins)) to include this new param

**Comment by acdvorak** [comment]
Thanks for the PR!

I'm not sure if I would consider `mdc-typography-adjust-margin` deprecated, because it's fairly common to want to change only the margin of an element (especially a heading). Let me double check with the team, though.

**Comment by pndewit** [comment]
Fair enough and thanks for the review :)

File: packages/mdc-typography/_mixins.scss (hunk 3 above)
Comment (pndewit) [reply packages/mdc-typography/_mixins.scss]: done

**Comment by pndewit** [comment]
@acdvorak is everything ok now?

**Comment by pndewit** [comment]
@acdvorak Can this be merged now?

**Comment by acdvorak** [comment]
Hey @pndewit, thanks for your patience, and sorry for the delay.

I talked it over with the team, and we would be happy to merge this PR with one small change: Remove the `$adjust-margin` param and always call `map-remove($style-props, margin)`.

Instead of creating a new param, you can just add the `mdc-typography--adjust-margin` class to any element that needs to set a margin, or call the `mdc-typography-adjust-margin` mixin.

I realize it might feel a little cumbersome to need to specify two separate CSS classes or call two separate mixins, but this separation allows us to A) use the `mdc-typography--*` classes on both UI elements (e.g., buttons) _and_ body text, and B) avoid "parameter creep" and mixin overlap.

**Comment by pndewit** [comment]
I understand, updated readme and removed argument.

**Comment by acdvorak** [comment]
LGTM! Thanks @pndewit!

**Pull request merged**